crates/astria-sequencer/src/app.rs

```diff
@@ -0,0 +1,195 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use penumbra_component::Component;
+use penumbra_storage::{
+    ArcStateDeltaExt,
+    Snapshot,
+    StateDelta,
+    Storage,
+};
+use tendermint::abci::{
+    self,
+    Event,
+};
+use tracing::info;
+
+use crate::{
+    accounts::component::AccountsComponent,
+    state_ext::{
+        StateReadExt as _,
+        StateWriteExt as _,
+    },
+    transaction::Transaction,
+};
+
+/// The application hash, used to verify the application state.
+/// TODO: this may not be the same as the state root hash?
+pub type AppHash = penumbra_storage::RootHash;
+
+/// The inter-block state being written to by the application.
+type InterBlockState = Arc<StateDelta<Snapshot>>;
+
+/// The genesis state for the application.
+#[derive(Clone, Debug, serde::Deserialize, Default)]
+pub struct GenesisState {
+    pub accounts: Vec<(String, u64)>,
+}
+
+/// The Sequencer application, written as a bundle of [`Component`]s.
+#[derive(Clone)]
+pub struct App {
```

Comment:
I kind of like `App` since it's the application logic :p this seems to be how it's referred to in ABCI as well. I think `Sequencer` should refer to the binary logic (ie what's run inside `main` like you mentioned in a previous comment?)